deployment-detector/src/agent.spec.ts

```diff
@@ -1,74 +1,96 @@
+      .setTo(mockDeployerAddress)
+      .addTraces({
+        function: FORTA_CREATE_AGENT,
+        to: "0xB50d3960a49120D0A6B543E7295cAE6C78d07967",
+        arguments: [mockarg.agentId, mockarg.owner, mockarg.metaData, mockarg.chainIds],
+      });
+    const findings: Finding[] = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
 
-    it("returns a finding if there is a Tether transfer over 10,000", async () => {
-      const mockTetherTransferEvent = {
-        args: {
-          from: "0xabc",
-          to: "0xdef",
-          value: ethers.BigNumber.from("20000000000"), //20k with 6 decimals
-        },
-      };
-      mockTxEvent.filterLog = jest
-        .fn()
-        .mockReturnValue([mockTetherTransferEvent]);
+  it("ignore transactions when correct deployer address calls a non-Forta contract that emits the same event", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(mockDeployerAddress)
+      .setTo(mockFortaContract)
+      .addTraces({
+        function: FORTA_CREATE_AGENT,
+        to: mockOtherFortaContract,
+        arguments: [mockarg.agentId, mockarg.owner, mockarg.metaData, mockarg.chainIds],
+      });
+    const findings: Finding[] = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
 
-      const findings = await handleTransaction(mockTxEvent);
+  it("creates an alert when a bot is deployed from the correct deployer address and the correct contract", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(mockDeployerAddress)
+      .setTo(mockFortaContract)
+      .addTraces({
+        function: FORTA_CREATE_AGENT,
+        to: mockFortaContract,
+        arguments: [mockarg.agentId, mockarg.owner, mockarg.metaData, mockarg.chainIds],
+      });
+    const findings: Finding[] = await handleTransaction(txEvent);
+    const mockFinding = [createFindingTest(mockarg.agentId, mockarg.owner, mockarg.chainIds)];
+    expect(findings).toStrictEqual(mockFinding);
+  });
 
-      const normalizedValue = mockTetherTransferEvent.args.value.div(
-        10 ** TETHER_DECIMALS
-      );
-      expect(findings).toStrictEqual([
-        Finding.fromObject({
-          name: "High Tether Transfer",
-          description: `High amount of USDT transferred: ${normalizedValue}`,
-          alertId: "FORTA-1",
-          severity: FindingSeverity.Low,
-          type: FindingType.Info,
-          metadata: {
-            to: mockTetherTransferEvent.args.to,
-            from: mockTetherTransferEvent.args.from,
-          },
-        }),
-      ]);
-      expect(mockTxEvent.filterLog).toHaveBeenCalledTimes(1);
-      expect(mockTxEvent.filterLog).toHaveBeenCalledWith(
-        ERC20_TRANSFER_EVENT,
-        TETHER_ADDRESS
-      );
-    });
+  it("ignore an alert when a bot is updated from the correct deployer address and the correct contract", async () => {
```

Comment:
Pardon if I didn't explain it well, but I meant that the bot should also catch cases where it updates a bot.